docs/sources/flow/reference/components/loki.server.md

```diff
@@ -0,0 +1,86 @@
+---
+title: loki.server
+---
+
+# loki.server
+
+`loki.server` is a common block used in all `loki.` components that expose a network server, such as
+[loki.source.gcplog][] and [loki.source.heroku][]. The
+network server supports both `HTTP` and `gRPC` protocols.
+
+[loki.source.gcplog]: {{< relref "./loki.source.gcplog.md" >}}
+[loki.source.heroku]: {{< relref "./loki.source.heroku.md" >}}
+
+## Usage
+
+```river
+server {
+    http {
+        listen_port = 8080
+        listen_adress = "0.0.0.0"
+    }
+    grpc {
+        listen_port = 8081
+    }
+}
+```
+
+## Arguments
+
+`loki.server` supports the following arguments:
+
+ Name                        | Type           | Description                               | Default | Required 
+-----------------------------|----------------|-------------------------------------------|---------|----------
+ `graceful_shutdown_timeout` | `RelabelRules` | Relabeling rules to apply on log entries. | "{}"    | no       
```

Comment:
This should be removed, right? There's no top-level arguments defined on the block.